tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -90,17 +114,29 @@ def parse_datafile_lines(datafile, program_type, section):
             records = parse_multi_record_line(line, schema)
 
             record_number = 0
-            for r in records:
+            for r, s in zip(records, schema.schemas):
                 record_number += 1
                 record, record_is_valid, record_errors = r
                 if not record_is_valid:
                     line_errors = errors.get(line_number, {})
-                    line_errors[record_number] = record_errors
-                    errors[line_number] = line_errors
+                    line_errors.update({record_number: record_errors})
+                    errors.update({line_number: record_errors})
+                store_record(unsaved_records, record, s.model)
         else:
-            record_is_valid, record_errors = parse_datafile_line(line, schema)
+            record, record_is_valid, record_errors = parse_datafile_line(line, schema)
             if not record_is_valid:
-                errors[line_number] = record_errors
+                errors.update({line_number: record_errors})
+            store_record(unsaved_records, record, schema.model)
+
+        if line_number % 50000 == 0 and header_count > 0:
```

Comment:
Not a fan of this 50k magic number floating here. Could the batch size be regulated inside one of our functions either the "store records" or "bulk create"? Maybe once the dict in store records hits our variable `batch_size`, we save it? That might be slightly more readable.